model/src/main/kotlin/io/spine/validation/bound/NumericBoundParser.kt

```diff
@@ -0,0 +1,203 @@
+            "The `($optionName)` option could not parse the `$number` bound value specified for" +
+                    " `${field.qualifiedName}` field. The value is out of range for the field" +
+                    " type `${field.type.name}` the option is applied to."
+        }
+
+        return KotlinNumericBound(parsed!!, exclusive)
+    }
+
+    private fun NumericOptionMetadata.parseFieldReference(
+        fieldPath: String,
+        exclusive: Boolean
+    ): KotlinNumericBound {
+        Compilation.check(fieldPath != field.name.value, file, field.span) {
+            "The `($optionName)` option cannot use `$fieldPath` field as a bound value for" +
+                    " the `${field.qualifiedName}` because self-referencing is prohibited." +
+                    " Please use other message fields."
+        }
+
+        val boundFieldPath = io.spine.base.fieldPath {
+            fieldName.addAll(fieldPath.split("."))
+        }
+
+        val boundField = try {
+            val (messageType, _) = typeSystem.findMessage(field.declaringType)!!
+            typeSystem.resolve(boundFieldPath, messageType)
+        } catch (e: IllegalStateException) {
+            Compilation.error(file, field.span) {
+                "The `($optionName)` option could not parse the `$fieldPath` field path specified" +
+                        " for `${field.qualifiedName}` field. Please make sure the provided field" +
+                        " path is valid: `${e.message}`."
+            }
+        }
+
+        val boundFieldType = boundField.type.primitive
+        Compilation.check(boundFieldType in numericPrimitives, file, field.span) {
+            "The `($optionName)` option cannot use `$fieldPath` field as a bound value for" +
+                    " the `${field.qualifiedName}` field due to its type" +
+                    " `${boundFieldType.name}`. Only singular numeric fields are supported."
+        }
+
+        return KotlinNumericBound(boundFieldPath, exclusive)
+    }
+
+    private companion object {
+        private val INTEGER = Regex("[-+]?\\d+")
+        private val FLOAT = Regex("[-+]?\\d+\\.\\d+([eE][-+]?\\d+)?")
+    }
+}
+
+/**
+ * Parses this [String] to a [Float], returning null if the string is not
+ * a valid finite float.
+ */
+private fun String.toFiniteFloatOrNull(): Float? =
+    toFloatOrNull()?.takeIf { it.isFinite() }
+
+/**
+ * Parses this [String] to a [Double], returning null if the string is not
+ * a valid finite double.
+ */
+private fun String.toFiniteDoubleOrNull(): Double? =
+    toDoubleOrNull()?.takeIf { it.isFinite() }
+
+
+/**
+ * Tells whether this [String] bound value contains a field reference.
+ */
+private fun String.isFieldReference() = first().run {
+    isLetter() || this == '_'
+}
+
+private fun unexpectedPrimitiveType(primitiveType: PrimitiveType): Nothing =
+    error(
+        "`KotlinNumericBound` cannot be created for `$primitiveType` field type." +
```

Comment:
Is it still `KotlinNumericBound`? Can we have it as just `NumericBound`, please?